rancher2/import_rancher2_catalog_v2.go

```diff
@@ -1,17 +1,20 @@
 package rancher2
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func resourceRancher2CatalogV2Import(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+func resourceRancher2CatalogV2Import(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	clusterID, name := splitID(d.Id())
 	d.Set("cluster_id", clusterID)
 	d.Set("name", name)
 
-	err := resourceRancher2CatalogV2Read(d, meta)
-	if err != nil || d.Id() == "" {
-		return []*schema.ResourceData{}, err
+	diag := resourceRancher2CatalogV2Read(ctx, d, meta)
+	if diag.HasError() || d.Id() == "" { // TODO - Check this if, it can break, didn't change the logic,.
```

Comment:
After talking with you I noticed that It could panic. 

I changed that Logic to : 

```
diag := resourceRancher2CatalogV2Read(ctx, d, meta)
	if diag.HasError() {
		return []*schema.ResourceData{}, errors.New(diag[0].Summary)
	}
	if d.Id() == "" {
		return []*schema.ResourceData{}, nil
	}

```

That way it matches the old behavior.

Reply:
Oh yeah, I see that in the new code/later commits. Lgtm thank you